applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_element_utilities.cpp

```diff
@@ -8,20 +8,166 @@
 //					 Kratos default license: kratos/license.txt
 //
 //  Main authors:    Philipp Bucher
+//                   Vicente Mataix Ferrandiz
+//                   Riccardo Rossi
+//                   Ruben Zorrilla
 //
 
 // System includes
 
 // External includes
 
 // Project includes
+#include "includes/checks.h"
 #include "structural_mechanics_element_utilities.h"
 #include "structural_mechanics_application_variables.h"
 #include "utilities/math_utils.h"
 
 namespace Kratos {
 namespace StructuralMechanicsElementUtilities {
 
+int BaseElementCheck(
+    const Element* pElement,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    const auto& r_geometry = pElement->GetGeometry();
+    const SizeType number_of_nodes = r_geometry.size();
+    const SizeType dimension = r_geometry.WorkingSpaceDimension();
+
+    // Verify that the variables are correctly initialized
+    KRATOS_CHECK_VARIABLE_KEY(DISPLACEMENT)
+    KRATOS_CHECK_VARIABLE_KEY(VELOCITY)
+    KRATOS_CHECK_VARIABLE_KEY(ACCELERATION)
+    KRATOS_CHECK_VARIABLE_KEY(DENSITY)
+    KRATOS_CHECK_VARIABLE_KEY(VOLUME_ACCELERATION)
+    KRATOS_CHECK_VARIABLE_KEY(THICKNESS)
+
+    // Check that the element's nodes contain all required SolutionStepData and Degrees of freedom
+    for ( IndexType i = 0; i < number_of_nodes; i++ ) {
+        const NodeType &rnode = r_geometry[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISPLACEMENT,rnode)
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(VELOCITY,rnode)
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(ACCELERATION,rnode)
+
+        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_X, rnode)
+        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Y, rnode)
+        KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Z, rnode)
+    }
+
+    // Verify that the constitutive law exists
+    KRATOS_ERROR_IF_NOT(pElement->GetProperties().Has( CONSTITUTIVE_LAW )) << "Constitutive law not provided for property " << pElement->GetProperties().Id() << std::endl;
+
+    // Verify that the constitutive law has the correct dimension
+    const SizeType strain_size = pElement->GetProperties().GetValue( CONSTITUTIVE_LAW )->GetStrainSize();
+    if ( dimension == 2 ) {
+        KRATOS_ERROR_IF( strain_size < 3 || strain_size > 4) << "Wrong constitutive law used. This is a 2D element! expected strain size is 3 or 4 (el id = ) " << pElement->Id() << std::endl;
+    } else {
+        KRATOS_ERROR_IF_NOT(strain_size == 6) << "Wrong constitutive law used. This is a 3D element! expected strain size is 6 (el id = ) "<<  pElement->Id() << std::endl;
+    }
+
+    return 0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ComputeEquivalentF(
+    const Element* pElement,
+    Matrix& rF,
+    const Vector& rStrainTensor
```

Comment:
I would first put all input then all output
I think this is an unwritten convention

=> switch the order of rF and rStrainTensor